workqueue.c

```diff
@@ -20,6 +21,8 @@ enum {
 	SW_F_ERROR	= 1 << 4,
 };
 
+static pthread_mutex_t serialize_lock;
```

Comment:
This should be per workqueue struct, not a global entity. I'm also suspecting that this type of alloc is why it doesn't work for processes, just threads?